setup.py

```diff
@@ -27,12 +27,12 @@
 #------------------------------------------------------------------------------
 
 from setuptools import setup
-import re
+import re, io
 
 # setup.py shall not import adal
 __version__ = re.search(
     r'__version__\s*=\s*[\'"]([^\'"]*)[\'"]',  # It excludes inline comment too
-    open('adal/__init__.py').read()).group(1)
+    io.open('adal/__init__.py', encoding='utf8').read()).group(1)
```

Comment:
The right encoding to deal with BOM is `utf-8-sig`, but guess it doesn't matter here as we are not consuming the whole file content, rather just let `re` find a specific portion. The leading `0xefbb` will not fail the match anyway.